src/resource/worker-control.c

```diff
@@ -1082,12 +1095,31 @@ static int _on_worker_proxy_idle_timeout_event(sid_resource_event_source_t *es,
 
 static int _on_worker_signal_event(sid_resource_event_source_t *es, const struct signalfd_siginfo *si, void *arg)
 {
-	sid_resource_t *res = arg;
-
-	log_debug(ID(res), "Received signal %d from %d.", si->ssi_signo, si->ssi_pid);
-	sid_resource_exit_event_loop(res);
+    sid_resource_t *res = arg;
+    sid_resource_t *worker_res;
+    struct worker  *worker;
+
+    log_debug(ID(res), "Received signal %d from %d.", si->ssi_signo, si->ssi_pid);
+
+    switch (si->ssi_signo) {
+        case SIGTERM:
+        case SIGINT:
+            sid_resource_exit_event_loop(res);
+            break;
+        case SIGUSR1:
+            if (sid_resource_match(res, &sid_resource_type_worker, NULL))
+                worker_res = res;
+            else if (!(worker_res = sid_resource_search(res, SID_RESOURCE_SEARCH_ANC, &sid_resource_type_worker, NULL)))
+                break;
```

Comment:
This check is probably not needed because we have `_on_worker_signal_event` always called with the `worker_res` passed in...

Reply:
Ok - I see.  Thanks - removed.